src/init.cpp

```diff
@@ -348,7 +349,6 @@ void DynamicDataLoader::unload_data()
     SNIPPET.clear_snippets();
     vehicle_prototype::reset();
     vpart_info::reset();
-    MonsterGenerator::generator().reset();
```

Comment:
Yes, [here](https://github.com/CleverRaven/Cataclysm-DDA/pull/17093/files#diff-18513665750ef5adf42b5ec29e14162eR323). This line was moved because of the problem I mentioned both in the description and the comment: singletons should be destroyed independently, so that order doesn't matter. With this line here, the `game` class tries to reset the already destroyed object.